app/src/main/java/org/dhis2/usescases/searchTrackEntity/mapView/SearchTEMap.kt

```diff
@@ -0,0 +1,262 @@
+package org.dhis2.usescases.searchTrackEntity.mapView
+
+import android.content.Context
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.fragment.app.activityViewModels
+import com.mapbox.mapboxsdk.geometry.LatLng
+import com.mapbox.mapboxsdk.maps.MapboxMap
+import java.io.File
+import javax.inject.Inject
+import org.dhis2.animations.CarouselViewAnimations
+import org.dhis2.commons.bindings.clipWithRoundedCorners
+import org.dhis2.commons.resources.ColorUtils
+import org.dhis2.databinding.FragmentSearchMapBinding
+import org.dhis2.maps.ExternalMapNavigation
+import org.dhis2.maps.carousel.CarouselAdapter
+import org.dhis2.maps.layer.MapLayerDialog
+import org.dhis2.maps.managers.TeiMapManager
+import org.dhis2.maps.model.MapStyle
+import org.dhis2.usescases.general.FragmentGlobalAbstract
+import org.dhis2.usescases.searchTrackEntity.SearchTEActivity
+import org.dhis2.usescases.searchTrackEntity.SearchTEContractsModule
+import org.dhis2.usescases.searchTrackEntity.SearchTEIViewModel
+import org.dhis2.usescases.searchTrackEntity.SearchTeiViewModelFactory
+import org.dhis2.utils.NetworkUtils
+import org.dhis2.utils.customviews.ImageDetailBottomDialog
+
+const val ARG_FROM_RELATIONSHIP = "ARG_FROM_RELATIONSHIP"
+const val ARG_TE_TYPE = "ARG_TE_TYPE"
+
+class SearchTEMap : FragmentGlobalAbstract(), MapboxMap.OnMapClickListener {
+
+    @Inject
+    lateinit var mapNavigation: ExternalMapNavigation
+
+    @Inject
+    lateinit var presenter: SearchTEContractsModule.Presenter
```

Comment:
Could the presenter by used by the ViewModel instead and maintain only  View - ViewModel relationship?